### PR TITLE
OptionalUtil

### PR DESCRIPTION
**Pull request by beatngu13**
*Before submission, please check that ...*

- [x] the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.
- [x] the necessary tests are either created or updated.
- [x] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [ ] you updated the changelog.
- [ ] you updated necessary documentation within [retest/docs](https://github.com/retest/docs).

## Description

Rename `StreamUtil` to `OptionalUtil` as this is essentially `Optional#stream()` from Java 9, and add `or` helper also from Java 9.

For more details, please refer to the commit messages.

### CODE REVIEW

File: src/main/java/de/retest/recheck/util/OptionalUtil.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package de.retest.recheck.util;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+public class OptionalUtil {
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/util/OptionalUtil.java:7]: If we only could use [`@ExtensionMethod`](https://projectlombok.org/features/experimental/ExtensionMethod)

File: src/main/java/de/retest/recheck/util/OptionalUtil.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+	private OptionalUtil() {}
+
+	@SuppressWarnings( "unchecked" )
+	public static <T> Stream<T> stream( final Optional<? extends T> o ) {
+		return (Stream<T>) o.map( Stream::of ).orElseGet( Stream::empty );
+	}
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/util/OptionalUtil.java:14]: Do we want to use a real method and not a lambda?

```java
private static final Function<Optional<?>, Stream<?>> toStream = o -> o.map( Stream::of ).orElse( Stream.empty() );

public static <T> Function<? super Optional<T>, Stream<T> stream() {
    return /*Fancy cast here*/ toStream;
}
```

File: src/test/java/de/retest/recheck/util/OptionalUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	@Test
+	void stream_should_be_empty_if_absent() throws Exception {
+		final Optional<Object> absent = Optional.empty();
+		assertThat( OptionalUtil.stream( absent ) ).isEmpty();
+	}
+
+	@Test
+	void or_should_return_given_optional_if_present() throws Exception {
+		final Optional<Object> opt = Optional.of( new Object() );
+		final Supplier<Optional<Object>> supplier = () -> Optional.empty();
```
Comment (diba1013) [inline src/test/java/de/retest/recheck/util/OptionalUtilTest.java]: Do you want the same scheme as below with `absent` and `present`

**Review by diba1013 — CHANGES_REQUESTED** [review]
Next PR "upgrade to java 11": The desperate cry ...

File: src/main/java/de/retest/recheck/util/OptionalUtil.java (hunk 1 above)
Comment (beatngu13) [reply src/main/java/de/retest/recheck/util/OptionalUtil.java:7]: Or [Kotlin](https://kotlinlang.org/docs/reference/extensions.html). 😉 But yes, I vote against experimental Lombok features in prod.

File: src/main/java/de/retest/recheck/util/OptionalUtil.java (hunk 2 above)
Comment (beatngu13) [reply src/main/java/de/retest/recheck/util/OptionalUtil.java:14]: Dunno, what is the benefit?

File: src/test/java/de/retest/recheck/util/OptionalUtilTest.java (hunk 3 above)
Comment (beatngu13) [reply src/test/java/de/retest/recheck/util/OptionalUtilTest.java]: Like:

```java
final Optional<Object> present = Optional.of( new Object() );
final Supplier<Optional<Object>> absent = () -> Optional.empty();
assertThat( OptionalUtil.or​( present, absent ) ).isEqualTo( present );
```

?

File: src/main/java/de/retest/recheck/util/OptionalUtil.java (hunk 2 above)
Comment (diba1013) [reply src/main/java/de/retest/recheck/util/OptionalUtil.java:14]: Maybe increased readability on site of the usage and&mdash;on a lower note&mdash;fewer objects created?

```java
list.stream() //
    .map( OptionalUtil::stream )

list.stream() //
    .map( OptionalUtil.stream() )
```

But same as the discussion about complex lambda vs method reference within a static method: No real preference.

File: src/test/java/de/retest/recheck/util/OptionalUtilTest.java (hunk 3 above)
Comment (diba1013) [reply src/test/java/de/retest/recheck/util/OptionalUtilTest.java]: Yes

File: src/main/java/de/retest/recheck/util/OptionalUtil.java (hunk 2 above)
Comment (beatngu13) [reply src/main/java/de/retest/recheck/util/OptionalUtil.java:14]: I think I personally prefer the current, method-ref-based style. But thanks for bringing attention to this.

**Review by diba1013 — APPROVED** [review]

**Pull request merged**